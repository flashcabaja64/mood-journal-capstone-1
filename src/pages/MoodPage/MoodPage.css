.mood-page {
  margin: 0 auto;
  min-width: 100vw;
  display: flex;
  flex-wrap: wrap;
}

/* Title of the 'My Moods' tab page */
.mood-page>h1 {
  display: block;
  font-size: 3rem;
  padding: 40px 0 10px 0;
  margin: 0 auto;
}

/* Search title and input field */
h3.search-title {
  font-size: 20px;
  font-style: italic;
  padding-bottom: 0;
  display: block;
  margin-left: 20px;
}

input.mood-page-input {
  display: block;
  padding: 10px;
  margin-left: 20px;
  font-size: 15px;
  width: 10rem;
}
