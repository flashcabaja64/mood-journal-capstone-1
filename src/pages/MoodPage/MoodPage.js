import React, { Component } from 'react';
//import CommentContext from '../../MoodContext/CommentContext'
import MoodContext from '../../MoodContext/MoodContext';
import EntryApiService from '../../services/entry-api-service'
import MoodListItem from '../../components/MoodListItem/MoodListItem';
import './MoodPage.css'

export default class MoodPage extends Component {
  // constructor(props) {
  //   super(props)
  // }

  state = {
    search: ''
  }
  static contextType = MoodContext;
  //static contextType = CommentContext;

  // Can possibly delete .then and .catch block
  componentDidMount() {
    this.context.clearError()
    EntryApiService.getEntries()
      .then(this.context.setEntryList)
      .catch(this.context.setError)
  }

  onSearch = e => {
    this.setState({ search: e.target.value })
  }

  // renderEntries() {
  //   const { entries = [] } = this.context
  //   return entries.map(entry =>
  //     <MoodListItem
  //       key={entry.id}
  //       id={entry.id}
  //       entry={entry}
  //     />
  //   )
  // }

  render() {
    const { entries = [] } = this.context
    const {search} = this.state
    let filterEntries = entries.filter(entry => entry.content === search)

    // Refactor this into a ternary operator
    if(search) {
      filterEntries = entries.filter(entry => {
        return entry.content.toLowerCase().indexOf(search.toLowerCase()) !== -1
      })
    } else {
      filterEntries = entries.map(entry => entry)
    }
    
    return(
      <>
        <h3 className="search-title">Search Entry Contents:</h3>
          <input
            className="mood-page-input"
            type="text"
            value={this.state.search}
            onChange={this.onSearch}
            placeholder="Search journal entry contents...">
          </input>
        <div className="mood-page">
        <h1>My Moods</h1>
        {filterEntries.map(entry =>
          <MoodListItem
            key={entry.id}
            id={entry.id}
            entry={entry}
        />)}
        </div>
        {/* {this.renderEntries()} */}
      </>
    )
  }
}