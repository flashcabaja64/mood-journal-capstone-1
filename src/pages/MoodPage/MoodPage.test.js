import React from 'react';
import ReactDOM from 'react-dom';
import MoodPage from './MoodPage'
import renderer from 'react-test-renderer';


it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(<MoodPage />, div)

  ReactDOM.unmountComponentAtNode(div)
})

it('renders UI as expected', () => {
  const tree = renderer
  .create(<MoodPage />)
  .toJSON();
  expect(tree).toMatchSnapshot();
})