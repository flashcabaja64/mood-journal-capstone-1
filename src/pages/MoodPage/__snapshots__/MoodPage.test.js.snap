// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI as expected 1`] = `
Array [
  <h3
    className="search-title"
  >
    Search Entry Contents:
  </h3>,
  <input
    className="mood-page-input"
    onChange={[Function]}
    placeholder="Search journal entry contents..."
    type="text"
    value=""
  />,
  <div
    className="mood-page"
  >
    <h1>
      My Moods
    </h1>
  </div>,
]
`;
